classes/user/Collector.php

```diff
@@ -455,7 +455,7 @@ protected function buildUserGroupFilter(Builder $query): self
                 )
                 ->when($this->excludeRoles !== null, fn ($query) => $query->whereNotIn('ug.role_id', $this->excludeRoles))
                 ->when($this->roleIds !== null, fn ($query) => $query->whereIn('ug.role_id', $this->roleIds))
-                ->when($this->contextIds !== null, fn ($query) => $query->whereIn('ug.context_id', $this->contextIds))
+                ->when($this->contextIds !== null, fn ($query) => $query->whereIn(DB::raw('COALESCE(ug.context_id, 0)'), $this->contextIds))
```

Comment:
We're going to run into a collision here between the conventions we use for "this applies to the site level rather than the journal level" (for which we previously used `CONTEXT_ID_NONE` but are changing to `null`) and "there is no filter on this condition", for which in the new DAO/Collector patterns we use `null`.

What you're implying in this change is that `$this->contextIds` should be `null` when no filter is set, and should be an array containing `0` when we want to query site-wide user groups (e.g. the administrator user group). (This is workable, though I'd prefer an array containing `null`, as set out below.)

However, in other cases that don't use the new query builder toolset, you're moving away from `CONTEXT_ID_NONE` and instead preferring `null` directly (e.g. [this change](https://github.com/pkp/pkp-lib/pull/9076/files#diff-5d0146e413cc69e673f0e04a928ef6ac5eadac70bd11dd9af3c0e45f89ff4826R74)).

This makes it ambiguous whether coders should use `null` or `CONTEXT_ID_NONE`. As we convert more DAOs over to the `Collector` pattern it'll get muddier.

Do you think it's feasible to code consistently around `null`s rather than a mix of `null` and `0`, and change the `CONTEXT_ID_NONE` constant so that it's synonymous with `null`?

This will also reduce the use of `COALESCE` in the SQL, which risks proliferating and might have repercussions on performance if they start to spread into complex queries.